actions/rm-runner-ip/entrypoint.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -eu
+
+# remove the runner IP from the list of authorized IPs
+CUR_IPS=$(gcloud container clusters describe "${K8_CLUSTER}" --format json | \
+    jq '[ .masterAuthorizedNetworksConfig.cidrBlocks[] | .cidrBlock ]')
+RUNNER_IP=$(curl 'https://api.ipify.org/?format=text')
+RESTORE_IPS=$(echo "${CUR_IPS}" | \
+    jq -r --arg ip "${RUNNER_IP}" '[ .[] | select(startswith($ip) | not) ] | unique | join(",")')
+
+# restore the original list of authorized IPs if they exist
+gcloud container clusters update "${K8_CLUSTER}" \
```

Comment:
It occurs to me that this process is subject to a race condition. Between the point where the old list is captured and the new list is set, another process could also be modifying the list. If that occurs, then you could both have the case where a newly added IP is removed, or a newly deleted IP is added back in.

I don't know how likely this problem is, and fixing it would require some kind of locking, which might be a lot of work to implement, so I don't have a feeling for how important this issue is.

Reply:
Oh, does the `lock-namespace` action prevent this? If so, 👍

Reply:
No, the `lock-namespace` action does not prevent this. You're correct there's a race condition, but I don't know a good solution to fix this.